shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -231,9 +235,11 @@ - (void)setSystemChromeSystemUIOverlayStyle:(NSDictionary*)message {
                       object:nil
                     userInfo:@{@(kOverlayStyleUpdateNotificationKey) : @(statusBarStyle)}];
   } else {
+#if !APPLICATION_EXTENSION_API_ONLY
     // Note: -[UIApplication setStatusBarStyle] is deprecated in iOS9
     // in favor of delegating to the view controller
     [[UIApplication sharedApplication] setStatusBarStyle:statusBarStyle];
```

Comment:
`setStatusBarStyle` was deprecated in iOS 9 so we wouldn't even need a fallback or the macro (minimum is now iOS 11), if we can get a view controller here (`[_engine.get() viewController]`?)
```objc
- (void)setStatusBarStyle:(UIStatusBarStyle)statusBarStyle animated:(BOOL)animated API_DEPRECATED("Use -[UIViewController preferredStatusBarStyle]", ios(2.0, 9.0)) API_UNAVAILABLE(tvos);

```
Though I don't really understand the `delegateToViewController` logic or why we wouldn't want to pass it along to the embedder... #2732